docs/deeplearning4j/templates/build-from-source.md

```diff
@@ -74,148 +91,152 @@ Homebrew is the accepted method of installing prerequisite software. Assuming yo
+2. [Visual Studio](https://visualstudio.microsoft.com/) - Based on the CUDA SDK version, you'll need different versions of Visual Studio.
+
+Visit the following link to confirm which version of Visual Studio is supported with the installed CUDA SDK:
+`https://docs.nvidia.com/cuda/archive/`<CUDA_VERSION>`/cuda-toolkit-release-notes/index.html#cuda-compiler-new-features`
+
+Replace `<CUDA_VERSION>` with your installed CUDA_SDK version in the link above. For example, https://docs.nvidia.com/cuda/archive/10.0/cuda-toolkit-release-notes/index.html#cuda-compiler-new-features
+                                                                                              
+In order to build the CUDA backend you will have to setup some more environment variables first, by calling `vcvars64.bat`.
+But first, set the system environment variable `SET_FULL_PATH` to `true`, so all of the variables that `vcvars64.bat` sets up, are passed to the mingw shell.
+Additionally, you need to open the `mingw64.ini` in your msys64 installation folder and add the command: `MSYS2_PATH_TYPE=inherit`. After that, follow the steps below:
 
-After that, you should see a lot of activity and installs on the terminal. To verify that you have, for example, *gcc*, enter this line:
+1. In your visual studio installation folder, look for the file `vcvars64.bat`. After it's found, go to its location and run it. This will set up some environment variables and open `cmd.exe`.
+2. Inside the console, run `c:\msys64\mingw64.exe`. This will open up the MSYS2 shell with the environment variables set so, you can execute the required maven commands to build the DL4J stack for the GPU architecture.
 
-    gcc --version
+See the section below for the required build commands.
 
-For more complete instructions, [go here](http://www.cyberciti.biz/faq/centos-linux-install-gcc-c-c-compiler/).
+## Build Instructions
 
-##### OS X
 
-You can install OpenBLAS on OS X with Home
-Science:
+First clone the DL4J repository and navigate to the `deeplearning4j` folder 
 
+```bash
+git clone https://github.com/deeplearning4j/deeplearning4j.git
+cd deeplearning4j
 ```
-brew install homebrew/science/openblas
-```
-
-##### Windows
 
-An OpenBLAS package is available for `msys2`. You can install it using the `pacman` command.
+If you want to change the Scala, Spark or Cuda versions, you can execute one or more of the following commands:
+```bash
+./change-cuda-versions.sh x.x
+./change-scala-versions.sh 2.xx
+./change-spark-versions.sh x
+```
 
-#### ATLAS
+Now, to build for each architecture, execute the relevant commands:
 
-##### Linux
+### Building for CPU
 
-**Ubuntu**
-An apt package is available for ATLAS on Ubuntu:
+Let's first set some variables for ease:
 
-```
-sudo apt-get install libatlas-base-dev libatlas-dev
+```bash
+PLATFORM=<YOUR_PLATFORM> # Can be either one of [linux-x86_64, macosx-x86_64, windows-x86_64]
```

Comment:
This is a very short list of platforms compared to the one at [this issue comment](https://github.com/deeplearning4j/deeplearning4j/issues/6787#issuecomment-442747638) of @sshepel .
Probably, there are some platforms that need some additional building instructions, but I guess that there are some more than just [linux-x86_64, macosx-x86_64, windows-x86_64] that we can build with following the current documentation.
Where can we find a list of supported platforms? (If I understand well, these are the accepted values for `-Dlibnd4j.platform` and `-Djavacpp.platform`.)